cmd/configschema/fields.go

```diff
@@ -21,8 +21,27 @@ import (
 	"time"
 
 	"github.com/fatih/structtag"
+	"go.uber.org/multierr"
 )
 
+func ConfigsToFields(dr DirResolver, configs []CfgInfo) (map[string]map[string]*Field, error) {
+	m := map[string]map[string]*Field{}
+	var errs error
+	for _, cfg := range configs {
+		fields, err := ReadFields(reflect.ValueOf(cfg.CfgInstance), dr)
```

Comment:
Agree that generics are a better option than reflection (if it's possible to use them instead), but here we need reflection because we are getting the fields of a struct at runtime.